[gajinkee] IP

gajinkee opened the pull request:
# Skynet PR Update 🚀
> "Do it now." - Anonymous

## Summary
This PR introduces important changes to Skynet, enhancing its chat capabilities and performance.

## Features Added:
- Improved message encryption 🔒
- Optimized server response time :hourglass: 
- Added support for **Markdown messages**
- Bug fixes and performance improvements

## Steps to Test:
1. Download from here [skynet](https://github.com/gajinkee/ip)
2. Download the latest release
3. Run with 'java -jar skynet.jar' 

~~Paid~~
Completely **FREE**!

Features:
- [X] Managing tasks
- [X]   Managing deadlines (coming soon)
- [ ]    Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gajinkee committed:
level-0 done

gajinkee committed:
Level-1 done

gajinkee committed:
Completed level 2

gajinkee committed:
level 3 with some level A-classes

gajinkee committed:
Added Task.java class

gajinkee committed:
TED's added to tasks

gajinkee committed:
Fixed Inheritance bugs

gajinkee committed:
TextUITesting done

gajinkee committed:
level 5 fin

gajinkee committed:
Added the Exception class for missing arguments

gajinkee committed:
did level 6 and collections was already done.

gajinkee committed:
Completed Level 6 and some bug fixes

gajinkee committed:
Main: Add Enums

gajinkee committed:
storage: Added local storage

gajinkee committed:
Merge branch 'branch-Level-7'

gajinkee committed:
DeadLineTask: Added dateTime recognition

gajinkee committed:
Merge branch 'branch-Level-8'

gajinkee committed:
Skynet.java: Refactored and repackage code

gajinkee committed:
Merge branch 'branch-A-MoreOOP'

gajinkee committed:
Merge branch 'add-gradle-support' of https://github.com/gajinkee/ip into add-gradle-support

gajinkee committed:
Merge branch 'add-gradle-support'

gajinkee committed:
JUnit: Added gradle and unit tests

gajinkee committed:
JavaDocs: Added JavaDocs to public classes/methods

gajinkee committed:
Parser: Find method included

gajinkee committed:
Merge branch 'branch-A-JavaDoc'

gajinkee committed:
Style: Minor style changes

gajinkee committed:
Merge branch 'branch-A-CodingStandard'

gajinkee committed:
find: find method added and conflicts resolved

gajinkee committed:
Style: Fix style for all classes

gajinkee committed:
javaFX: Added GUI

gajinkee committed:
Merge branch 'branch-Level-10'

gajinkee committed:
CheckStyle: Fix styling issues

waylonggggg commented on src/main/java/skynet/ui/Parser.java:
```diff
@@ -0,0 +1,251 @@
+     * @param taskList array of tasks used
+     * @param ui       The interface for input and output
+     */
+    public static void handleCommand(TaskList taskList, UI ui) {
```
Should the code for this method be shorter? Perhaps helper functions could be used to shorten the code.

waylonggggg commented on src/main/java/skynet/ui/Parser.java:
```diff
@@ -0,0 +1,251 @@
+     * @param inputLine Input given by the user.
+     * @return Response String given by Skynet.
+     */
+    public static String handleGuiCommand(TaskList taskList, UI ui, String inputLine) {
```
Perhaps the method could be condensed using helper functions?

waylonggggg reviewed (commented):
LGTM. Just a few nits to fix

gajinkee committed:
Add assertions as part of a multi-part PR

Appropriate assertions placed in Parser to check no null values appear.
Future changes might introduce null values which should be checked.

As a step for future commits, assertions are added for testing in the
development phase.

gajinkee committed:
Refactor Parser.

Commit is part of a multi-commit PR. This commit is
mean't for the refactoring of code to meet code quality.

Parser has been refactored into Command Enum previously
named UserInput for the sake of code quality.

gajinkee committed:
Merge pull request #1 from gajinkee/branch-A-Assertions

Add assertions as part of a multi-part PR

gajinkee committed:
Merge branch 'master' into branch-A-CodeQuality

Conflict resolution for multi-part PR

Mulit-part PR resulted in many conflicts. All conflicts were resolved
and application restored to working condition.

gajinkee committed:
Merge pull request #2 from gajinkee/branch-A-CodeQuality

Refactor Parser

gajinkee committed:
Task: remove unused methods

gajinkee committed:
TaskList: Handle duplicate tasks

gajinkee committed:
TaskList: Add javaDoc comments

gajinkee committed:
Merge branch 'branch-C-DetectDuplicates'

gajinkee committed:
Workflow: GitHub actions CI

gajinkee committed:
Doc: add user guide

gajinkee committed:
CodeQuality: improve code quality

gajinkee committed:
Checkstyle: fix checkstyle
